services/movie.py

```diff
@@ -0,0 +1,44 @@
+from db.models import Movie
+from django.db.models import QuerySet
+
+
+def get_movies(
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None
+) -> QuerySet:
+    queryset = Movie.objects.all()
+    if genres_ids and actors_ids:
+        return queryset.filter(
+            genres__id__in=genres_ids
+        ).filter(
+            actors__id__in=actors_ids
+        )
+
+    if genres_ids:
+        return queryset.filter(genres__id__in=genres_ids)
+
+    if actors_ids:
+        return queryset.filter(actors__id__in=actors_ids)
+
+    return queryset
+
+
+def get_movie_by_id(movie_id: int) -> object:
+    return Movie.objects.get(id=movie_id)

```

Comment:
if object doesn't exist u will get an internal server error, if u don't handle it in the middleware u should use
get_object_or_404() func instead